### ISSUE TITLE
Physics body scaling could not be applied.

### ISSUE THREAD

**Issue by bnolan** [c-frame/aframe-extras#22]
Heya, what's the plan with physics body scaling. I'm getting it with an entity that is a box that I'm scaling. It's easy for me to refactor the code to set the width height and depth in this instance, but it'd be nice to support scaling for scenes that are created in the editor. I could write a PR that manually scales elements by calling multiplyScalar on all the vertices? That'll be expensive if the scale is constantly changing, but should be fine for bodies that are scaled on creation but otherwise don't change.

**Comment by donmccurdy** [comment]
I'm comfortable with the assumption that scale won't change constantly. Skipped it and added the warning when I noticed some of the Cannon shape instances had a scaling method and others didn't. Would gladly take a PR, or will get to it when i can thanks!

**Comment by bnolan** [comment]
I'll have a crack at the PR :)

On Monday, 25 April 2016, Don McCurdy notifications@github.com wrote:

> I'm comfortable with the assumption that scale won't change constantly.
> Skipped it and added the warning when I noticed some of the Cannon shape
> instances had a scaling method and others didn't. Would gladly take a PR,
> or will get to it when i can thanks!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/donmccurdy/aframe-extras/issues/22#issuecomment-214592094

## 

Regards,
Ben Nolan

https://www.scenevr.com/

twitter: bnolan
cell: +64 21 770 662

**Comment by donmccurdy** [comment]
This should be fixed now, on the [aframe-v0.3.0 branch](https://github.com/donmccurdy/aframe-extras/tree/aframe-v0.3.0) – I no longer rely on the `scale` attribute directly, but instead apply each object's matrix recursively to its children. Scale on a wrapper entity like this won't be applied:

``` html
<a-entity scale="2 2 2">
  <a-entity obj-model="obj: url(...)" dynamic-body></a-entity>
</<a-entity>
```

... but that could be fixed by using `matrixWorld` if needed. Which is probably a better longterm solution anyway.

**Comment by donmccurdy** [comment]
Should be more thoroughly fixed by 0ccb8630dcb6cc1277a132ab04c6bd378f8de223.
